sdk/storage/storage-blob/review/storage-blob.api.md

```diff
@@ -574,6 +574,12 @@ export type BlobDownloadResponseModel = BlobDownloadHeaders & {
     };
 };
 
+// @public
+export interface BlobDownloadResponseParsed extends BlobDownloadResponseModel {
+    objectReplicationDestinationPolicyId?: string;
+    objectReplicationSourceProperties?: ObjectReplicationPolicy[];
```

Comment:
Can there be two different policies with the same policy id? If not, making this `Record<string, ObjectReplicationRule[]>` would likely be easier on everyone.

Reply:
I don't think there will be two different policies with the same policy id. I agree a dictionary is more intuitive and .NET first went with it but was told to use list. I will ask Sean about the reason.

Reply:
Quoting Kamil:
>There are few reasons why, mostly protecting ourselves from breaks to accommodate unforeseen feature development:
> - List is simpler representation over dictionary and yet delivers same data to customer, it doesn’t make any assumptions about keys/ids, what is unique what not. Thus offers more room. Customer can convert the list to dictionary if they want to – this is one-liner with modern collection APIs in any language. And as you said the cardinality of policies is expected to be low, so conversion won’t be compute heavy.
> - What if Policy gains new properties? List<Policy> can handle that easily without break. Dictionary<string->Rule> can’t. We’d need to introduce new models to accommodate this
> - What if policyId becomes ambiguous (this is less likely but who knows) , i.e. someone introduces policy versions OR set of rules can change by time window. Dictionary makes strong assumption it doesn’t change. List does not, therefore is open for this extension as well without any break.

Reply:
> List is simpler representation over dictionary

In what sense?

> this is one-liner with modern collection APIs in any language

This is not an easy one-liner for JS at least. Moreover, if the idea is "well we might have duplicates in the future", such code is likely to be bugged (and serves as strong rationale for why policy IDs must not be duplicated).

> What if Policy gains new properties? 

Whether it's Rule[] or Record<id, Rule>, I don't see why adding/removing properties is different in either case. Can you clarify?

> Dictionary makes strong assumption it doesn’t change.

I don't understand this point either, is this a .NET thing?

Reply:
The original idea came from API Review for .NET please see https://apiview.dev/Assemblies/Review/a17b6175b93847db83a99c7fcfeca9ab/a8aab056477a4a2f87e8b1cd14ac7fe2 (@tg-msft  FYI). and evolved into current form.

Reply:
My feedback was primarily that `Dictionary<string, Dictionary<string, string>>` isn't very helpful in C# because properties named `Key` and `Value` don't tell me as much as `PolicyId` or `Rule`.  I agree with Brian's point that an array is less pleasant to lookup values with the standard library in JS and there's no reason we couldn't have a `Dictionary<string, ObjectReplicationSourcePolicy>` where `Key == ObjectReplicationSourcePolicy.PolicyId`.

Reply:
Do we expect that major use case is going to be customer checking status with policy-id that has been acquired from somewhere else? My guess is that prevailing use case is going to be one (few) policy and customer enumerating them to check the status, mostly taking always first element. On the other hand dictionary could be enumerated as well. Given that id most probably won't become ambiguous ever we could do that and change to dictionary.
However, in principle, I think exposing dictionary is more commitment for us than a simple list.